lib/sdk/oauth2-flows/AuthCodeAbstract.ts

```diff
@@ -19,254 +19,254 @@ import type {
+	 */
+	protected getCallbackURLParams(callbackURL: URL): [string, string] {
+		const searchParams = new URLSearchParams(callbackURL.search);
+		const state = searchParams.get('state')!;
+		const error = searchParams.get('error');
+		const code = searchParams.get('code')!;
+
+		if (error) {
+			throw new Error(`Authorization server reported an error: ${error}`);
+		}
+
+		return [code, state];
+	}
+
+	/**
+	 * Method implements logic for fetching tokens from the authorization server using
+	 * the provided body, the `useCookies` is used exclusively on the browser.
+	 * @param {SessionManager} sessionManager
+	 * @param {URLSearchParams} body
+	 * @param {boolean} useCookies
+	 * @returns {Promise<OAuth2CodeExchangeResponse>}
+	 */
+	protected async fetchTokensFor(
+		sessionManager: SessionManager,
+		body: URLSearchParams,
+		useCookies: boolean = false
+	): Promise<OAuth2CodeExchangeResponse> {
+		const headers = new Headers();
+		headers.append(
+			'Content-Type',
+			'application/x-www-form-urlencoded; charset=UTF-8'
+		);
+		headers.append(
+			...getSDKHeader({
+				frameworkVersion: this.config.frameworkVersion,
+				framework: this.config.framework,
+			})
+		);
+
+		const config: RequestInit = {
+			method: 'POST',
+			headers,
+			body,
+			credentials: useCookies ? 'include' : undefined,
+		};
+		const response = await fetch(this.tokenEndpoint, config);
+		const payload = (await response.json()) as
+			| OAuth2CodeExchangeErrorResponse
+			| OAuth2CodeExchangeResponse;
+
+		const errorPayload = payload as OAuth2CodeExchangeErrorResponse;
+		if (errorPayload.error) {
+			await sessionManager.destroySession();
+			const errorDescription = errorPayload.error_description;
+			const message = errorDescription ?? errorPayload.error;
+			throw new Error(message);
+		}
+
+		return payload as OAuth2CodeExchangeResponse;
+	}
+
+	/**
+	 * Helper method employed by @see {createAuthorizationURL} method above for
+	 * generating the aforementioned authorization URL.
+	 * @param {AuthURLOptions}
+	 * @returns {URLSearchParams}
+	 */
+	protected generateAuthURLParams(options: AuthURLOptions = {}): URLSearchParams {
+		const searchParams = this.getBaseAuthURLParams();
+
+		let searchParamsObject: Record<string, string> = {
+			scope: this.config.scope ?? AuthCodeAbstract.DEFAULT_TOKEN_SCOPES,
+		};
+
+		if (this.config.audience) {
+			searchParamsObject.audience = this.config.audience;
+		}
+
+		if (options.start_page) {
+			searchParamsObject.start_page = options.start_page;
+		}
+
+		if (options.org_code) {
+			searchParamsObject.org_code = options.org_code;
+		}
+
+		if (options.is_create_org) {
+			searchParamsObject.org_name = options.org_name ?? '';
+			searchParamsObject.is_create_org = 'true';
+		}
+
+		if (options.post_login_redirect_url) {
+			searchParamsObject.post_login_redirect_url = options.post_login_redirect_url;
+		}
+
+		if (options.authUrlParams) {
+			searchParamsObject = Object.assign(options.authUrlParams, searchParamsObject);
```

Comment:
@peterphanouvong , we might want to spread the properties onto a new object, this will result in `options.authUrlParams` being modified, which might not be expected.